internal/handlers/handler_oauth2_consent.go

```diff
@@ -293,61 +293,17 @@ func handleOAuth2ConsentFlowIDPOST(ctx *middlewares.AutheliaCtx, bodyJSON oidc.C
-			if client.GetConsentPolicy().Mode == oidc.ClientConsentModePreConfigured {
-				config := model.OAuth2ConsentPreConfig{
-					ClientID:      consent.ClientID,
-					Subject:       consent.Subject.UUID,
-					CreatedAt:     ctx.Clock.Now(),
-					ExpiresAt:     sql.NullTime{Time: ctx.Clock.Now().Add(client.GetConsentPolicy().Duration), Valid: true},
-					Scopes:        consent.GrantedScopes,
-					Audience:      consent.GrantedAudience,
-					GrantedClaims: bodyJSON.Claims,
-				}
-
-				var (
-					requests *oidc.ClaimsRequests
-				)
-
-				if requests, err = oidc.NewClaimRequests(form); err != nil {
-					ctx.Logger.
-						WithError(err).
-						WithFields(map[string]any{logging.FieldFlowID: consent.ChallengeID.String(), logging.FieldUsername: userSession.Username, logging.FieldClientID: consent.ClientID, logging.FieldSessionID: consent.ID}).
-						Error("Error occurred parsing request form claims parameter")
-				} else if requests == nil {
-					config.RequestedClaims = sql.NullString{Valid: false}
-					config.SignatureClaims = sql.NullString{Valid: false}
-				} else if config.RequestedClaims.String, config.SignatureClaims.String, err = requests.Serialized(); err != nil {
-					ctx.Logger.
-						WithError(err).
-						WithFields(map[string]any{logging.FieldFlowID: consent.ChallengeID.String(), logging.FieldUsername: userSession.Username, logging.FieldClientID: consent.ClientID, logging.FieldSessionID: consent.ID}).
-						Error("Error occurred calculating claims signature for consent")
-				} else {
-					config.RequestedClaims.Valid = true
-					config.SignatureClaims.Valid = true
-				}
-
-				var id int64
-
-				if id, err = ctx.Providers.StorageProvider.SaveOAuth2ConsentPreConfiguration(ctx, config); err != nil {
-					ctx.Logger.
-						WithError(err).
-						WithFields(map[string]any{logging.FieldFlowID: consent.ChallengeID.String(), logging.FieldUsername: userSession.Username, logging.FieldClientID: consent.ClientID, logging.FieldSessionID: consent.ID}).
-						Error("Error occurred saving consent pre-configuration to the database")
-
-					ctx.SetJSONError(messageOperationFailed)
-
-					return
-				}
-
-				consent.PreConfiguration = sql.NullInt64{Int64: id, Valid: true}
-
+			switch {
+			case oidc.FormRequiresExplicitConsent(form):
 				ctx.Logger.
-					WithFields(map[string]any{logging.FieldFlowID: consent.ChallengeID.String(), logging.FieldUsername: userSession.Username, logging.FieldClientID: consent.ClientID, logging.FieldSessionID: consent.ID, logging.FieldExpiration: config.ExpiresAt.Time.Unix()}).
-					Debug("Saved consent pre-configuration with expiration")
-			} else {
+					WithFields(map[string]any{logging.FieldFlowID: consent.ChallengeID.String(), logging.FieldUsername: userSession.Username, logging.FieldClientID: consent.ClientID, logging.FieldSessionID: consent.ID}).
+					Warn("Ignored saving pre-configuration as it is not permitted due to constraints within the authorization request from")
```

Comment:
```suggestion
					Warn("Ignored saving pre-configuration as it is not permitted due to constraints within the authorization request form")
```

I believe this is a typo.